failed: DnsDiscoverySpec

https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/14156

```
[DEBUG] [12/06/2018 13:51:56.096] [pool-1-thread-1] [EventStream(akka://AsyncDnsSystem)] logger log1-Logging$DefaultLogger started
[DEBUG] [12/06/2018 13:51:56.096] [pool-1-thread-1] [EventStream(akka://AsyncDnsSystem)] Default Loggers started
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: No such file or directory
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {}->unix://localhost:80
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: No such file or directory
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {}->unix://localhost:80
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: No such file or directory
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {}->unix://localhost:80
[ERROR] [12/06/2018 13:51:57.293] [pool-1-thread-1] [akka.actor.ActorSystemImpl(DnsDiscoverySpec)] Test not run as docker is not available
[12-06 12:51:57.298] [info] DnsDiscoverySpec:
Coroner Thread Count starts at 20 in akka.discovery.dns.DnsDiscoverySpec
[INFO] [12/06/2018 13:51:57.308] [pool-1-thread-1] [DnsDiscoverySpec(akka://DnsDiscoverySpec)] Running on port port 37519
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: No such file or directory
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {}->unix://localhost:80
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: No such file or directory
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {}->unix://localhost:80
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: No such file or directory
Dec 06, 2018 1:51:57 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {}->unix://localhost:80
[WARN] [12/06/2018 13:51:57.340] [pool-1-thread-1-ScalaTest-running-DnsDiscoverySpec] [DnsDiscoverySpec(akka://DnsDiscoverySpec)] Failed to pull docker image [raboof/bind:9.11.3-20180713-nochown], is docker running?
[12-06 12:51:57.342] [info] Dns Discovery with isolated resolver
[12-06 12:51:57.347] [info] Dns Discovery with isolated resolver (pending)
[12-06 12:51:57.348] [info] Dns discovery with the system resolver
[DEBUG] [12/06/2018 13:51:57.352] [pool-1-thread-1-ScalaTest-running-DnsDiscoverySpec] [DnsServiceDiscovery(akka://AsyncDnsSystem)] using system resolver as it is set to async-dns
[DEBUG] [12/06/2018 13:51:57.370] [pool-1-thread-1-ScalaTest-running-DnsDiscoverySpec] [DnsServiceDiscovery(akka://AsyncDnsSystem)] Lookup [Lookup(foo.test.,Some(service),Some(tcp))] translated to SRV query [_service._tcp.foo.test.] as contains portName and protocol
[DEBUG] [12/06/2018 13:51:57.396] [AsyncDnsSystem-akka.actor.default-dispatcher-4] [akka://AsyncDnsSystem/system/IO-DNS/async-dns/$a] Using name servers [List(localhost/127.0.0.1:37519)]
[DEBUG] [12/06/2018 13:51:57.400] [AsyncDnsSystem-akka.actor.default-dispatcher-6] [TimerScheduler(akka://AsyncDnsSystem)] Start timer [CacheCleanup] with generation [1]
[DEBUG] [12/06/2018 13:51:57.404] [AsyncDnsSystem-akka.actor.default-dispatcher-6] [akka://AsyncDnsSystem/system/IO-DNS] Resolution request for _service._tcp.foo.test. Srv from Actor[akka://AsyncDnsSystem/temp/$a]
[DEBUG] [12/06/2018 13:51:57.415] [AsyncDnsSystem-akka.actor.default-dispatcher-4] [akka://AsyncDnsSystem/system/IO-DNS/async-dns/$a] Attempting to resolve _service._tcp.foo.test. with Actor[akka://AsyncDnsSystem/system/IO-DNS/async-dns/$a/$a#1281120076]
[DEBUG] [12/06/2018 13:51:57.460] [AsyncDnsSystem-akka.actor.default-dispatcher-3] [akka://AsyncDnsSystem/system/IO-UDP-FF/selectors/$a/0] Successfully bound to [/0:0:0:0:0:0:0:0:40842]
[DEBUG] [12/06/2018 13:51:57.462] [AsyncDnsSystem-akka.actor.default-dispatcher-4] [akka://AsyncDnsSystem/system/IO-DNS/async-dns/$a/$a] Bound to UDP address [/0:0:0:0:0:0:0:0:40842]
[DEBUG] [12/06/2018 13:51:57.467] [AsyncDnsSystem-akka.actor.default-dispatcher-4] [akka://AsyncDnsSystem/system/IO-DNS/async-dns/$a/$a] Resolving [_service._tcp.foo.test.] (SRV)
[DEBUG] [12/06/2018 13:51:57.483] [AsyncDnsSystem-akka.actor.default-dispatcher-4] [akka://AsyncDnsSystem/system/IO-DNS/async-dns/$a/$a] Message to [localhost/127.0.0.1:37519]: [Message(1,<QUERY,RD,SUCCESS>,List(Question(_service._tcp.foo.test.,RecordType(33,SRV),RecordClass(1,IN))),List(),List(),List())]
[12-06 12:52:02.402] [info] - must work with SRV records *** FAILED *** (5 seconds, 39 milliseconds)
[12-06 12:52:02.402] [info]   A timeout occurred waiting for a future to complete. Queried 51 times, sleeping 100 milliseconds between each query. (DnsDiscoverySpec.scala:99)
[12-06 12:52:02.402] [info]   org.scalatest.concurrent.Futures$FutureConcept$$anon$1:
[12-06 12:52:02.402] [info]   at org.scalatest.concurrent.Futures$FutureConcept.tryTryAgain$1(Futures.scala:538)
[12-06 12:52:02.402] [info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValueImpl(Futures.scala:550)
[12-06 12:52:02.402] [info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValueImpl$(Futures.scala:479)
[12-06 12:52:02.402] [info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValueImpl(ScalaFutures.scala:275)
[12-06 12:52:02.402] [info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValue(Futures.scala:476)
[12-06 12:52:02.402] [info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValue$(Futures.scala:475)
[12-06 12:52:02.402] [info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValue(ScalaFutures.scala:275)
[12-06 12:52:02.402] [info]   at akka.discovery.dns.DnsDiscoverySpec.$anonfun$new$6(DnsDiscoverySpec.scala:99)
```

Looks like docker was not available:
```
[ERROR] [12/06/2018 13:51:57.293] [pool-1-thread-1] [akka.actor.ActorSystemImpl(DnsDiscoverySpec)] Test not run as docker is not available
```
We could also skip this test when docker is not available, but I'd prefer to fix the jenkins machines so docker is available again ;)

--- a/akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala
+++ b/akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala
@@ -90,6 +90,11 @@ class DnsDiscoverySpec extends AkkaSpec(DnsDiscoverySpec.config)
   }
 
   "Dns discovery with the system resolver" must {
+    if (!dockerAvailable()) {
+      system.log.error("Test not run as docker is not available")
+      pending
+    }
+    
     "work with SRV records" in {
       val discovery = Discovery(systemWithAsyncDnsAsResolver).discovery
       val name = "_service._tcp.foo.test."

--- a/akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala
+++ b/akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala
@@ -53,7 +53,7 @@ class DnsDiscoverySpec extends AkkaSpec(DnsDiscoverySpec.config)
   "Dns Discovery with isolated resolver" must {
 
     if (!dockerAvailable()) {
-      system.log.error("Test not run as docker is not available")
+      info("Test not run as docker is not available")
       pending
     }
 
@@ -91,10 +91,10 @@ class DnsDiscoverySpec extends AkkaSpec(DnsDiscoverySpec.config)
 
   "Dns discovery with the system resolver" must {
     if (!dockerAvailable()) {
-      system.log.error("Test not run as docker is not available")
+      info("Test not run as docker is not available")
       pending
     }
-    
+
     "work with SRV records" in {
       val discovery = Discovery(systemWithAsyncDnsAsResolver).discovery
       val name = "_service._tcp.foo.test."

